canonical-web-and-design/ubuntu.com
# Advantage shop product selection

## Done

- Added product selection markup and "waterfall" form logic to /advantage/subscribe

**Note:** clicking "Buy" won't trigger the payment modal, that will be covered in another PR that implements the shop payment endpoints

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/advantage/subscribe
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Use the form to make a product selection, see that at the end of it you are able to add a product to the cart
- Add the product to the cart
- Change quantities on the added product, see that the price and totals updates accordingly
- See that the header has been updated to "Any more to add?"
- Run through the process again, this time making different selections
  - Clicking AWS or Azure should disable the rest of the form
  - Clicking "Other" in the Ubuntu version should disable the rest of the form
  - Adding a second, different product should add a new line item to the cart
  - Adding multiples of the same product should update the existing line item with the combined quantity of both selections
- Remove products from the cart, see that totals are updated
- Remove all products from the cart, see that the cart disappears and the page header "Subscribe to Ubuntu Advantage" reappears, and the heading on the first section reads "What are you setting up?"
